expr/func_movingwindow.go

```diff
@@ -87,15 +96,15 @@ func (s *FuncMovingWindow) Exec(dataMap DataMap) ([]models.Series, error) {
 	return outputs, nil
 }
 
-func aggregateOnWindow(serie models.Series, aggFunc batch.AggFunc, interval uint32, xFilesFactor float64) (uint32, []schema.Point) {
+func aggregateOnWindow(serie models.Series, aggFunc batch.AggFunc, interval uint32, xFilesFactor float64) (uint32, []schema.Point, error) {
 	out := pointSlicePool.Get().([]schema.Point)
 
 	numPoints := len(serie.Datapoints)
 	serieStart, serieEnd := serie.QueryFrom, serie.QueryTo
 	queryStart := uint32(serieStart + interval)
 
 	if queryStart < serieStart {
-		panic("Query start is before the first point in series")
+		return 0, nil, errors.New("query start time cannot be before the first point in series")
```

Comment:
Do we have any idea what could cause this? Should this be marked as an Internal server error?

Reply:
It's just a sanity check. All the calculations that lead to this would be ours. So it's not an internal error

Reply:
That sounds exactly like an internal server error

Reply:
putting on the "user hat", that makes sense.